librz/core/cmd/cmd_analysis.c

```diff
@@ -8290,3 +7842,367 @@ RZ_IPI RzCmdStatus rz_analysis_basic_block_find_paths_handler(RzCore *core, int
+	ut64 timeout = rz_config_get_i(core->config, "analysis.timeout");
+	char *debugger = NULL;
+	ut64 old_offset = core->offset;
+	const char *notify = "Analyze all flags starting with sym. and entry0 (aa)";
+	rz_core_notify_begin(core, "%s", notify);
+	rz_cons_break_push(NULL, NULL);
+	rz_cons_break_timeout(timeout);
+	rz_core_analysis_all(core);
+	rz_core_notify_done(core, "%s", notify);
+	rz_core_task_yield(&core->tasks);
+	if (type == CORE_ANALYSIS_SIMPLE || rz_cons_is_breaked()) {
+		goto finish;
+	}
+	// Run pending analysis immediately after analysis
+	// Usefull when running commands with ";" or via rizin -c,-i
+	debugger = core->dbg->cur ? strdup(core->dbg->cur->name) : strdup("esil");
+	if (core->io && core->io->desc && core->io->desc->plugin && !core->io->desc->plugin->isdbg) {
+		// set it only if is debugging
+		RZ_FREE(debugger);
+	}
+	rz_cons_clear_line(1);
+	rz_core_analysis_everything(core, type == CORE_ANALYSIS_EXPERIMENTAL, debugger);
+finish:
+	rz_core_seek(core, old_offset, true);
+	// XXX this shouldnt be called. flags muts be created wheen the function is registered
+	rz_core_analysis_flag_every_function(core);
+	rz_cons_break_pop();
+	RZ_FREE(debugger);
+}
+
+// aa
+RZ_IPI RzCmdStatus rz_analyze_simple_handler(RzCore *core, int argc, const char **argv) {
+	core_perform_auto_analysis(core, CORE_ANALYSIS_SIMPLE);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aaa
+RZ_IPI RzCmdStatus rz_analyze_everything_handler(RzCore *core, int argc, const char **argv) {
+	core_perform_auto_analysis(core, CORE_ANALYSIS_DEEP);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aaaa
+RZ_IPI RzCmdStatus rz_analyze_everything_experimental_handler(RzCore *core, int argc, const char **argv) {
+	core_perform_auto_analysis(core, CORE_ANALYSIS_EXPERIMENTAL);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aac
+RZ_IPI RzCmdStatus rz_analyze_all_function_calls_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_analysis_calls(core, false);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aaci
+RZ_IPI RzCmdStatus rz_analyze_all_function_calls_to_imports_handler(RzCore *core, int argc, const char **argv) {
+	rz_core_analysis_calls(core, true);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aad
+RZ_IPI RzCmdStatus rz_analyze_all_data_references_to_code_handler(RzCore *core, int argc, const char **argv) {
+	RzListIter *iter;
+	RzAnalysisXRef *xref;
+	RzList *list = rz_analysis_xrefs_get_from(core->analysis, UT64_MAX);
+	rz_list_foreach (list, iter, xref) {
+		if (xref->type == RZ_ANALYSIS_REF_TYPE_DATA && rz_io_is_valid_offset(core->io, xref->to, false)) {
+			rz_core_analysis_fcn(core, xref->from, xref->to, RZ_ANALYSIS_REF_TYPE_NULL, 1);
+		}
+	}
+	rz_list_free(list);
+	return RZ_CMD_STATUS_OK;
+}
+
+// aaf
+RZ_IPI RzCmdStatus rz_analyze_all_functions_handler(RzCore *core, int argc, const char **argv) {
+	const bool old_hasnext = rz_config_get_b(core->config, "analysis.hasnext");
+	rz_config_set_b(core->config, "analysis.hasnext", true);
+	rz_core_cmd0(core, "afr@@c:isq"); // TODO: replace with C apis.
```

Comment:
should also there be a space before `@@`?

Reply:
it is not necessary though I'd place it anyhow for readability